### ISSUE TITLE
Syncthing Fork fails with Exit Code 2 when being started

### ISSUE THREAD

**Issue by glientsc** [Catfriend1/syncthing-android#656]
It is my first bug report...
Wanted to start Syncthing Fork again after a week or two and it always fails with exit code 2.
My phone: Samsung Galaxy Note 8 with custom rom N950FXXS7DSG1_Dr.Ketan_P10, root granted to syncthing fork with Magisk. Syncthing Fork version v1.5, probably I have not run it since the last update. On Windows I use Syncthing v1.5.0 64 bit, with SyncTrayzor
Here is the Log, I hope it helps
Kind regards, Guenther

18:44:07I/DecorView createDecorCaptionView >> DecorView@7e3cba8[], isFloating: false, isApplication: true, hasWindowDecorCaption: false, hasWindowControllerCallback: true
18:44:07I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:44:07W/libEGL   EGLNativeWindowType 0x79cf9f1010 disconnect failed
18:44:07W/libEGL   EGLNativeWindowType 0x79c27a7010 disconnect failed
18:44:35I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:44:35I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:44:35I/AssistStructure Flattened final assist data: 7480 bytes, containing 1 windows, 18 views
18:45:10I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:45:10W/IInputConnectionWrapper getExtractedText on inactive InputConnection
18:45:15W/libEGL   EGLNativeWindowType 0x79da3db010 disconnect failed
18:45:30I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:46:00I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:46:00W/IInputConnectionWrapper getExtractedText on inactive InputConnection
18:46:02I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:46:11I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:46:11I/AssistStructure Flattened final assist data: 7496 bytes, containing 1 windows, 18 views
18:46:47I/AssistStructure Flattened final assist data: 7536 bytes, containing 1 windows, 18 views
18:46:47I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:47:10I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:48:18I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:48:23W/libEGL   EGLNativeWindowType 0x79da3db010 disconnect failed
18:50:46I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:50:54I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:50:54W/libEGL   EGLNativeWindowType 0x79da3db010 disconnect failed
18:50:58I/DecorView createDecorCaptionView >> DecorView@6de118e[], isFloating: false, isApplication: true, hasWindowDecorCaption: false, hasWindowControllerCallback: true
18:50:58I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:50:58W/libEGL   EGLNativeWindowType 0x79cffcd010 disconnect failed
18:50:58W/libEGL   EGLNativeWindowType 0x79db746010 disconnect failed
18:51:07I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:51:07I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:51:07I/AssistStructure Flattened final assist data: 7480 bytes, containing 1 windows, 18 views
18:51:13I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:51:14I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:51:16I/AssistStructure Flattened final assist data: 6544 bytes, containing 1 windows, 17 views
18:51:23I/AssistStructure Flattened final assist data: 6560 bytes, containing 1 windows, 17 views
18:51:23I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:51:43I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:51:44I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:51:55I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:53:34W/libEGL   EGLNativeWindowType 0x79cdaac010 disconnect failed
18:53:34W/libEGL   EGLNativeWindowType 0x79cffd2010 disconnect failed
18:53:34W/libEGL   EGLNativeWindowType 0x79cda61010 disconnect failed
18:53:34I/yncthingandroi Explicit concurrent copying GC freed 268771(12MB) AllocSpace objects, 848(55MB) LOS objects, 49% free, 7MB/14MB, paused 186us total 109.032ms
18:53:35I/yncthingandroi Explicit concurrent copying GC freed 12603(584KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6MB/12MB, paused 66us total 43.885ms
18:56:34I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:56:36I/DecorView createDecorCaptionView >> DecorView@e778537[], isFloating: false, isApplication: true, hasWindowDecorCaption: false, hasWindowControllerCallback: true
18:56:37I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:56:37W/libEGL   EGLNativeWindowType 0x79c07fe010 disconnect failed
18:56:37W/libEGL   EGLNativeWindowType 0x79da3da010 disconnect failed
18:57:08I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:57:09I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:57:17I/AssistStructure Flattened final assist data: 7484 bytes, containing 1 windows, 17 views
18:57:23I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:57:24I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:57:28W/libEGL   EGLNativeWindowType 0x79cffb3010 disconnect failed
18:57:31I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:57:33W/libEGL   EGLNativeWindowType 0x79cffb3010 disconnect failed
18:57:41E/InputMethodManager prepareNavigationBarInfo() rootView is null
18:57:41I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:57:41I/yncthingandroi Explicit concurrent copying GC freed 35475(1644KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6MB/13MB, paused 73us total 48.106ms
18:57:41I/yncthingandroi Explicit concurrent copying GC freed 4583(266KB) AllocSpace objects, 0(0B) LOS objects, 50% free, 6MB/12MB, paused 70us total 43.903ms
18:57:42I/InputMethodManager startInputInner - mService.startInputOrWindowGainedFocus
18:57:50I/SyncthingService shouldRun decision changed to false according to configured run conditions.
18:57:51I/SyncthingService shouldRun decision changed to true according to configured run conditions.
18:57:52I/Util     runShellCommandGetOutput: Exited with code 0
18:57:52I/SyncthingService onServiceStateChange: from DISABLED to STARTING
18:57:52I/SyncthingService Web GUI will be available at https://127.0.0.1:8384
18:57:52I/Util     runShellCommandGetOutput: Exited with code 0
18:57:52I/PollWebGuiAvailableTask Starting to poll for web gui availability
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/SyncthingRunnable increaseInotifyWatches: sysctl returned 0
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/Util     runShellCommandGetOutput: Exited with code 0
18:57:52 [monitor] WARNING: Error starting the main Syncthing process: fork/exec /data/app/com.github.catfriend1.syncthingandroid-L6J2A9nDgBy_GVOXJfFqhA==/lib/arm64/libsyncthingnative.so: permission denied
18:57:52W/SyncthingNativeCode panic: Error starting the main Syncthing process
18:57:52W/SyncthingNativeCode 
18:57:52W/SyncthingNativeCode goroutine 1 [running]:
18:57:52W/SyncthingNativeCode main.monitorMain(0x400003c00c, 0x51, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/src/github.com/syncthing/syncthing/cmd/syncthing/monitor.go:125 +0x1128
18:57:52W/SyncthingNativeCode main.main()
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/src/github.com/syncthing/syncthing/cmd/syncthing/main.go:440 +0x42c
18:57:52W/SyncthingNativeCode 
18:57:52I/yncthingandroi Background concurrent copying GC freed 8204(2MB) AllocSpace objects, 564(24MB) LOS objects, 28% free, 58MB/82MB, paused 78us total 105.717ms
18:57:52W/SyncthingNativeCode goroutine 18 [syscall]:
18:57:52W/SyncthingNativeCode os/signal.signal_recv(0x0)
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android-prereq/go_1.13.9/src/runtime/sigqueue.go:147 +0xb8
18:57:52W/SyncthingNativeCode os/signal.loop()
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android-prereq/go_1.13.9/src/os/signal/signal_unix.go:23 +0x18
18:57:52W/SyncthingNativeCode created by os/signal.init.0
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android-prereq/go_1.13.9/src/os/signal/signal_unix.go:29 +0x30
18:57:52W/SyncthingNativeCode 
18:57:52W/SyncthingNativeCode goroutine 19 [chan receive]:
18:57:52W/SyncthingNativeCode github.com/syncthing/notify.(*nonrecursiveTree).dispatch(0x400011e420, 0x400011e360)
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/pkg/mod/github.com/syncthing/notify@v0.0.0-20190709140112-69c7a957d3e2/tree_nonrecursive.go:36 +0x98
18:57:52W/SyncthingNativeCode created by github.com/syncthing/notify.newNonrecursiveTree
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/pkg/mod/github.com/syncthing/notify@v0.0.0-20190709140112-69c7a957d3e2/tree_nonrecursive.go:29 +0xb8
18:57:52W/SyncthingNativeCode 
18:57:52W/SyncthingNativeCode goroutine 20 [chan receive]:
18:57:52W/SyncthingNativeCode github.com/syncthing/notify.(*nonrecursiveTree).internal(0x400011e420, 0x400011e3c0)
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/pkg/mod/github.com/syncthing/notify@v0.0.0-20190709140112-69c7a957d3e2/tree_nonrecursive.go:81 +0x34
18:57:52W/SyncthingNativeCode created by github.com/syncthing/notify.newNonrecursiveTree
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/pkg/mod/github.com/syncthing/notify@v0.0.0-20190709140112-69c7a957d3e2/tree_nonrecursive.go:30 +0xe0
18:57:52W/SyncthingNativeCode 
18:57:52W/SyncthingNativeCode goroutine 21 [sleep]:
18:57:52W/SyncthingNativeCode runtime.goparkunlock(...)
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android-prereq/go_1.13.9/src/runtime/proc.go:310
18:57:52W/SyncthingNativeCode time.Sleep(0x1dcd6500)
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android-prereq/go_1.13.9/src/runtime/time.go:105 +0x124
18:57:52W/SyncthingNativeCode github.com/syncthing/syncthing/lib/dialer.init.0.func2()
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/src/github.com/syncthing/syncthing/lib/dialer/internal.go:43 +0x24
18:57:52W/SyncthingNativeCode created by github.com/syncthing/syncthing/lib/dialer.init.0
18:57:52W/SyncthingNativeCode 	E:/GitHub/syncthing-android/syncthing/src/github.com/syncthing/syncthing/lib/dialer/internal.go:42 +0x184
18:57:52I/SyncthingRunnableIoNice ionice returned 127 on libsyncthingnative.so
18:57:52W/SyncthingRunnable exit reason = exitNoUpgradeAvailable. Another Syncthing instance may be already running.
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/SyncthingService onServiceStateChange: from ERROR to DISABLED
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:52I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:53I/Util     runShellCommandGetOutput: Exited with code 0
18:57:53W/ApiRequest Request to https://127.0.0.1:8384/rest/system/shutdown failed, java.net.ConnectException: Failed to connect to /127.0.0.1:8384
18:57:53I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:53I/System.out (HTTPLog)-Static: isSBSettingEnabled false
18:57:53I/yncthingandroi Background concurrent copying GC freed 5785(2MB) AllocSpace objects, 1364(74MB) LOS objects, 32% free, 50MB/74MB, paused 344us total 135.968ms
18:58:09I/DecorView createDecorCaptionView >> DecorView@3f032bd[], isFloating: false, isApplication: true, hasWindowDecorCaption: false, hasWindowControllerCallback: true

**Comment by Catfriend1** [comment]
Hi, thanks for reaching out. I'll see what I can do for help during my next dev session.

The problem should be here:
> 18:57:52 [monitor] WARNING: Error starting the main Syncthing process: fork/exec /data/app/com.github.catfriend1.syncthingandroid-L6J2A9nDgBy_GVOXJfFqhA==/lib/arm64/libsyncthingnative.so: permission denied
18:57:52W/SyncthingNativeCode panic: Error starting the main Syncthing process

**Comment by glientsc** [comment]
Oh, thank you for your extremely fast answer. Anything that I can do or try to help resolving this issue, I will. App permissions are all granted (so I am not sure which permission is denied)...

**Comment by Catfriend1** [comment]
If its possible for you, you could test if the native starts okay when root mode is turned off in the settings. Plus, it would be interesting for me if the previous release works: https://github.com/Catfriend1/syncthing-android/releases/download/v1.4.2.4/com.github.catfriend1.syncthingandroid_v1.4.2.4_615fcfcb.apk

**Comment by glientsc** [comment]
both a "yes". Having turned off root, it started fine (albeit it could not synchronize the folders on my sd-card anymore because of that). Turning on root while it was running, it crashed again.
I turned off sycthing fork, installed the previous apk you gave me, it took me through the initial setup process, everything works fine, even after switching on root. Of course, all the folder connections I set up in the official syncthing fork play store app are not taken over, so I did not do any testing in that regard. Thanks for your time!
